packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -639,6 +639,11 @@ function useRestoreFocus(scopeRef: RefObject<Element[]>, restoreFocus: boolean,
             let treeNode = clonedTree.getTreeNode(scopeRef);
             while (treeNode) {
               if (treeNode.nodeToRestore && document.body.contains(treeNode.nodeToRestore)) {
+                // in case of collections, the nodeToRestore may have changed, such as hovering over a new list item in submenus
+                // defer to the collection's focus scope to focus the correct element
+                if (treeNode.nodeToRestore.tabIndex === -1) {
```

Comment:
This seems like it might cause other problems. There are lots of reasons an element might have tabIndex="-1", and we might still want to restore focus to one, so I'm not sure we can make this assumption. Shouldn't we already not restore focus if it already moved elsewhere outside the scope? Above we are checking if the activeElement is the body. Is something there not working?

Reply:
the issue is described in depth here https://github.com/adobe/react-spectrum/pull/4192/commits/c5da3b0aab92d2328e2d2a82131dd22db8b2148e#diff-4c0c6856d378bccf3d5c99f8e69916bbb807602dfaf5da4e277f92d9288f5250R97